rfd/0128-aws-integration-tests.md

```diff
@@ -0,0 +1,191 @@
+---
+authors: Tiago Silva (tiago.silva@goteleport.com)
+state: draft
+---
+
+# RFD 128 - AWS End To End Tests
+
+## Required Approvers
+
+- Engineering: `@r0mant && @smallinsky`
+- Security: `@reedloden || @jentfoo`
+
+## What
+
+As part of the increase of the reliability of Teleport, we want to add integration
+tests for AWS. This will allow us to test the integration with AWS without having
+to rely on manual testing.
+
+## Why
+
+We want to increase the reliability of Teleport by adding integration tests for
+AWS. This will allow us to test the integration with AWS without having to rely
+on manual testing. This will also allow us to test the integration with AWS in
+a more reliable process to ensure we don't introduce regressions in the future
+or if we do, we can catch them early.
+
+Teleport integrates deeply with AWS to provide a seamless experience for users
+when using auto-discovery of nodes, databases and Kubernetes clusters. This
+integration is critical for the success of Teleport and we want to ensure that
+we can test it reliably.
+
+Each integration test will use the minimum required
+AWS API permissions to test the integration. This will ensure that we don't
+introduce regressions by changing the permissions required by Teleport but those
+changes are not detected because another test requires the same permissions.
+
+The goal of this RFD is to define the scope of the integration tests for AWS
```

Comment:
I would include that while we're focusing on AWS first, we should design the system with extending it to other clouds we support in mind.